### DISCUSSION TITLE
about the position of a isometric tile

### DISCUSSION THREAD

**Discussion by sparkok** [excaliburjs/Excalibur#2581] [Q&A]
I have an isometric tiled map,and its map tile is 64 x 32,and  the size of features in the map is 64 x 128. 
I want to make a red rectange to match the blue feature in the map,but I find there is a vertical offset,I don't understand what is wrong?
the code is following
```
   getLastIsoMapLayer(): IsometricMap {
        const layers = this.getIsoMapLayers();
        if(layers.length >0){
            return layers[layers.length-1];
        }
    }
```
```
    const lastIsoLayer = this.getLastIsoMapLayer()
    const c = 0,r = 0
    const tile = lastIsoLayer.getTile(c,r)
    const pos = lastIsoLayer.tileToWorld(vec(c,r))
    const play = new Player()
    play.pos = pos
    play.z = tile.get(TransformComponent).z - 0.1
    this.add(play)
```
the code of Player
```
export class Player extends Actor {
  tileWidth = 64
  tileHeight = 32
  graphic :Canvas
  constructor() {
    super({
      width: 64,
      height: 128,
      anchor: vec(32/64,(128-16)/128),
    });
  }
  onInitialize() {
    this.graphic = new Canvas({
      width: this.width,
      height: this.height,
      cache: true,  
      draw: (context) => {
          context.lineWidth=2.7;
          context.strokeStyle="#ff0000"

          context.moveTo(0, 0);
          context.lineTo(this.width,0);
          context.lineTo(this.width,this.height);
          context.lineTo(0,this.height);
          context.lineTo(0,0);
          context.stroke();

          context.moveTo(this.width/2, 0);
          context.lineTo(this.width/2,this.height);
          context.stroke();

          context.moveTo(0,this.height/2);
          context.lineTo(this.width,this.height/2);
          context.stroke();
      }
  })
    this.graphics.use(this.graphic);
  }
}
```
<img width="582" alt="1" src="https://user-images.githubusercontent.com/7651365/220497194-545c93ae-26e7-4d61-ba90-ed014dfa6449.png">
<img width="370" alt="2" src="https://user-images.githubusercontent.com/7651365/220497315-30cc8963-22f0-440d-945a-c9d68cabf198.png">

**Comment by eonarheim** [comment]
@sparkok I'll look into this today, do you have a sample repo you can share?

I suspect the difference might be due to how isometric tiles are drawn. To match Tiled, they draw from the bottom left corner of their graphics https://github.com/excaliburjs/Excalibur/blob/main/src/engine/TileMap/IsometricMap.ts#L178

I have a post that explains the reasoning better here https://erikonarheim.com/posts/handling-height-in-isometric/

It's possible that if you draw the player in the same manner it might match with what you expect, here is an example https://github.com/eonarheim/isometric-layers/blob/main/player.ts

**Reply by sparkok** [reply]
this is my .tmx file.
[question2.zip](https://github.com/excaliburjs/Excalibur/files/10807268/question2.zip)

**Reply by sparkok** [reply]
my local project is too complex to run easily，so i have to make a sample project and put it on github.

**Reply by eonarheim** [reply]
That would be a big help 👍

**Reply by sparkok** [reply]
hello,eonarheim,I made a repo.
https://github.com/sparkok/learn-excaliburjs1
start is with 'yarn run dev',and click the menu 'no matched',you can show the not matched result.
the error code is in the function (testPlaceTinInCenterOfTile in level-one.ts).
thank you!

**Reply by eonarheim** [reply]
@sparkok This is perfect, I'll dig into this 👍
